protocol.md

```diff
@@ -870,6 +902,13 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports $/partialResult notifications.
+	 * If true, the server is allowed to let the result in the final response be null
+	 * and only send the result through $/partialResult
+	 */
+	streaming?: boolean;
```

Comment:
> If a client and server supports streaming, it should work for all methods

If a client and server adds support for streaming for one method, do they get streaming for all methods for free? 

If there is is incremental work (on either/both client/server) to support each method then it would be be beneficial to allow a streaming toggle per method to lower the barrier to entry. In that case this could be a dictionary of method name to boolean, or we could add a flag to the ReferenceContext.